### PR TITLE
update build instructions for mac os

### PR DESCRIPTION
**Pull request by dwightguth**
Improves install instructions as well as performs various minor fixes to the build system.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,26 +69,21 @@ On ArchLinux:
 sudo pacman -S  base-devel rsync opam pandoc jre8-openjdk mpfr maven
 ```
```
Comment (msaxena2) [inline README.md]: I'm not sure you need to install XCode. You need just need the command line tools package, which most users would already have.

File: README.md [hunk 2]
```diff
@@ -69,26 +69,21 @@ On ArchLinux:
...
 
-On OSX, using [Homebrew](https://brew.sh/):
+On OSX, using [Homebrew](https://brew.sh/), after installing XCode:
```
Comment (msaxena2) [inline README.md]: I'm not sure you need to install XCode. You need just need the command line tools package, which most users would already have.

File: README.md [hunk 3]
```diff
@@ -69,26 +69,21 @@ On ArchLinux:
...
 brew tap caskroom/cask caskroom/version
 brew cask install java8
-brew install bison flex libtool autoconf gmp mpfr pandoc maven opam
-```
-
-Note that Homebrew install flex as a keg-only package. To install flex's library, use:
-
-```sh
-brew link --force flex
+brew install automake libtool gmp mpfr pkg-config pandoc maven opam python z3
```
Comment (msaxena2) [inline README.md]: OSX comes with python 2.7 pre-installed. Not sure you'd need to install it via homebrew.

File: README.md (hunk 2 above)
Comment (dwightguth) [reply README.md]: I will update this to mention the command line tools explicitly.

File: README.md (hunk 3 above)
Comment (dwightguth) [reply README.md]: I don't really know what to say except that python wasn't in the path when I built evm on a freshly upgraded high Sierra today, and this fixed it. Are you sure that's a correct statement?

**Comment by dwightguth** [comment]
@msaxena2 please re-review.

File: Makefile [hunk 4]
```diff
@@ -7,8 +7,8 @@
 all: build split-tests
 
 clean:
-	rm -r .build
-	find tests/proofs/ -name '*.k' -delete
+	rm -rf .build/java .build/ocaml .build/logs
+	-find tests/proofs/ -name '*.k' -delete
```
Comment (msaxena2) [inline Makefile]: Is this change to `find` => `-find` intentional/necessary?

Comment (dwightguth) [reply Makefile]: yes, find fails if tests/proofs doesn't exist, which it doesn't if you've just checked out the repository, so without this, the make clean target fails.

**Comment by msaxena2** [comment]
@dwightguth Changes look good, and we should get these in asap. But looks like the PR has gone beyond just updating the Mac related instructions. I think either @nishantjr or @ehildenb should review the changes to the makefile and ubuntu sections, since I didn't write them. Also, I see you're still working on the PR, but before we merge, we should clean up the commits. I can take care of that you'd like, and merge.

File: Makefile (hunk 4 above)
Comment (nishantjr) [reply Makefile]: Would `[ ! -d tests/proofs ] || find tests/proofs/ -name '*.k' -delete` be more explicit and easier to understand?

**Comment by ehildenb** [comment]
@dwightguth you can say "use `K_BIN` if set otherwise use `DEFAULT_VALUE`" by using `${K_BIN:-DEFAULT_VALUE}` in bash

File: kast-json.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
```
Comment (nishantjr) [inline kast-json.py]: Is this intended? On some systems where both `python2` and `python3` this can be a problem.

File: Makefile (hunk 4 above)
Comment (ehildenb) [reply Makefile]: I like @nishantjr's suggestion here.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Improve `make k-deps` a bit

**Comment by ehildenb** [comment]
ah I see, my bad

**Comment by dwightguth** [comment]
I will look at the test failures tomorrow morning

**Comment by MrChico** [comment]
I can't build this version and it seems like the problem is that the module `Constants` imported in KRYPTO.ml cannot be found. Running `make all` yields;
```
[Error] Compiler: Had 1 parsing errors.
[Error] Internal: gcc returned nonzero exit code. See output for details.
File ".build/ocaml/driver-kompiled/constants.ml", line 1:
Error: I/O error: .build/ocaml/driver-kompiled/constants.ml: No such file or directory
File "KRYPTO.ml", line 1, characters 5-14:
Error: Unbound module Constants
File "_none_", line 1:
Error: Cannot find file KRYPTO.cmx
```

**Comment by ehildenb** [comment]
@MrChico I think you need to run `make deps ; make all` now. The reason being that `make deps` always triggers a fetch from external sources, which is slow, so we wanted to separate it out into an explicit step.

**Comment by dwightguth** [comment]
@MrChico There is an issue in the makefile causing it to continue past the first error it reaches. The error is actually "gcc returned nonzero exit code". This is happening because we no longer require or support the homebrew keg-only version of flex being linked. Can you try `brew unlink flex` and let me know if it fixes the issue on this branch? That will be the supported behavior going forward.

**Comment by msaxena2** [comment]
Also, just to add onto to @dwightguth , you may want to check if you have flex (referred to as libl, not libfl on macs) in your linker's path. After unlinking flex via brew, try running "ld -ll" - that'd make sure you still have flex in your path.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by ehildenb** [comment]
@dwightguth commit messages

**Comment by dwightguth** [comment]
This is now ready for final review.

**Comment by nishantjr** [comment]
@dwightguth I force pushed a change that adds `z3` to ArchLinux install instructions.
Looks good to me.

**Comment by MrChico** [comment]
Builds like a charm. Thanks!

**Pull request merged**